Destination Pinecone: Add source_tag for attribution + unit tests

bindipankhudi opened the pull request:
Related tickets:
* https://github.com/airbytehq/airbyte/issues/38162
* https://github.com/airbytehq/airbyte/issues/38161

Changes in this PR:
* Pass a new param `source_tag` when initializing `PineconeGRPC` module. Tag is set to "airbyte" by default. For integration tests, tag is set to "airbyte_tests". Documentation for [reference](https://pinecone-2-partner-integration-guide.mintlify.app/integrations/build-integration/attribute-api-activity). 
* Unrelated to above: Added unit tests to bring test coverage to 90%

bindipankhudi committed:
add dummy test

bindipankhudi committed:
added attribution and related tests

bindipankhudi committed:
added new unit test

bindipankhudi committed:
chore: format code

bindipankhudi committed:
add test

bindipankhudi committed:
version increment and doc

bindipankhudi commented on airbyte-integrations/connectors/destination-pinecone/destination_pinecone/indexer.py:
```diff
@@ -25,14 +26,17 @@
 MAX_IDS_PER_DELETE = 1000
 
+AIRBYTE_TAG = "airbyte"
+AIRBYTE_TEST_TAG = "airbyte_test"
```
I thought it would be good to pass a tag in either case, but we can also not pass a tag during test. lmk what you think.

aaronsteers replied:
This is great! It's always good to make sure test invocations don't count towards (or against) user activity metrics.

bindipankhudi commented on airbyte-integrations/connectors/destination-pinecone/unit_tests/destination_test.py:
```diff
@@ -66,6 +66,26 @@ def test_check_with_errors(self, MockedEmbedder, MockedPineconeIndexer):
         mock_embedder.check.assert_called_once()
         mock_indexer.check.assert_called_once()
 
+    def test_check_with_config_errors(self):
```
unrelated to attribution change - these two tests are for improving test coverage.

aaronsteers commented on airbyte-integrations/connectors/destination-pinecone/integration_tests/pinecone_integration_test.py:
```diff
@@ -48,6 +50,8 @@ def tearDown(self):
                 print("Noting to delete. No data in the index/namespace.")
 
+    def test_integration_test_flag_is_set(self):
+        assert os.getenv("INTEGRATION_TEST", False) == "True"
```
```suggestion
        assert "PYTEST_CURRENT_TEST" in os.environ
```
 
TIL 🙂

https://stackoverflow.com/a/58866220/4298208

aaronsteers commented on airbyte-integrations/connectors/destination-pinecone/integration_tests/pinecone_integration_test.py:
```diff
@@ -32,6 +33,7 @@ def _wait(self):
         print()  # Move to the next line after the loop
     
     def setUp(self):
+        os.environ["INTEGRATION_TEST"] = "True"
```
```suggestion
```

I think we can use the built-in `PYTEST_CURRENT_TEST` env var.

 https://stackoverflow.com/a/58866220/4298208

aaronsteers commented on airbyte-integrations/connectors/destination-pinecone/destination_pinecone/indexer.py:
```diff
@@ -62,6 +66,10 @@ def pre_sync(self, catalog: ConfiguredAirbyteCatalog):
         return []
 
+    def get_source_tag(self):
+        is_integration_test = os.getenv("INTEGRATION_TEST", False)
```
```suggestion
        is_integration_test = "PYTEST_CURRENT_TEST" in os.environ
```

aaronsteers reviewed (changes requested):
This is great. One suggestion - which is totally optional - but I think we can use a built-in env var Rather than declaring a new one. This should be slightly more stable since new tests will also automatically have this set. 

One additional issue I'm seeing now is that CAT tests don't rely on PyTest, and we should try to account for them as well. 

Can we check if CAT and/or airbyte-ci set an enrollment variable which we can detect?

bindipankhudi committed:
Update airbyte-integrations/connectors/destination-pinecone/destination_pinecone/indexer.py

Co-authored-by: Aaron ("AJ") Steers <aj@airbyte.io>

bindipankhudi committed:
Update airbyte-integrations/connectors/destination-pinecone/integration_tests/pinecone_integration_test.py

Co-authored-by: Aaron ("AJ") Steers <aj@airbyte.io>

bindipankhudi committed:
Update airbyte-integrations/connectors/destination-pinecone/integration_tests/pinecone_integration_test.py

Co-authored-by: Aaron ("AJ") Steers <aj@airbyte.io>

bindipankhudi committed:
added check for CI env variable

bindipankhudi committed:
updated date in doc

bindipankhudi committed:
Merge remote-tracking branch 'origin/master' into bindi/destination-pinecone-test-coverage

bindipankhudi commented:
> This is great. One suggestion - which is totally optional - but I think we can use a built-in env var Rather than declaring a new one. This should be slightly more stable since new tests will also automatically have this set.
> 
> One additional issue I'm seeing now is that CAT tests don't rely on PyTest, and we should try to account for them as well.
> 
> Can we check if CAT and/or airbyte-ci set an enrollment variable which we can detect?

* Added the built in pytest env variable
* Added check for CI

bindipankhudi committed:
Merge branch 'master' into bindi/destination-pinecone-test-coverage

bindipankhudi commented:
@aaronsteers, addressed the review feedback. We are using built-in pytest env variable now, and also checking the CI env variable.

aaronsteers reviewed (approved):
Looks great!! 🚀🎉
